numpy/lib/arraysetops.py

```diff
@@ -448,8 +448,14 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     ar1 = np.asarray(ar1).ravel()
     ar2 = np.asarray(ar2).ravel()
 
-    # This code is significantly faster when the condition is satisfied.
-    if len(ar2) < 10 * len(ar1) ** 0.145:
+    # Check if one of the arrays is of type object
+    is_object = ar1.dtype == np.dtype('O') or ar2.dtype == np.dtype('O')
```

Comment:
I think you actually want `ar1.dtype..hasobject or ar2.dtype.hasobject`, to catch structured dtypes containing arrays

Reply:
Needs a test for this too

Reply:
thx for the hint!